src/mono/mono/metadata/monitor.c

```diff
@@ -1430,6 +1430,29 @@ mono_monitor_wait (MonoObjectHandle obj_handle, guint32 ms, MonoBoolean allow_in
 	return success;
 }
 
+#ifdef HOST_WASM
+void
+mono_set_string_interned_internal (MonoObject* obj)
+{
+	LockWord tmp;
+	tmp.sync = obj->synchronisation;
+	if (!lock_word_is_inflated (tmp))
+		mono_monitor_inflate (obj);
+	obj->synchronisation->is_interned = 1;
```

Comment:
synchronisation is a tagged pointer, this write is wrong. Use lock_word_get_inflated_lock